docs/proposals/20191017-kubeadm-based-control-plane.md

```diff
@@ -346,58 +376,72 @@ spec:
 - Completely removing the control plane and issuing a delete on the underlying machines.
 - User documentation should focus on deletion of the Cluster resource rather than the KubeadmControlPlane resource.
 
-##### KubeadmControlPlane rollout (using create-swap-and-delete)
+##### KubeadmControlPlane rollout
 
-- Triggered by: 
-    - Changes to Version
-    - Changes to the kubeadmConfigSpec
-    - Changes to the infrastructureRef
-    - The `upgradeAfter` field, which can be set to a specific time in the future
-        - Set to `nil` or the zero value of `time.Time` if no upgrades are desired
-        - An upgrade will run when that timestamp is passed
-        - Good for scheduling upgrades/SLOs
-        - Set `upgradeAfter` to now (in RFC3339 form) if an upgrade is required immediately
-
-- Rollout operations rely on scale up and scale down which are be blocked based on Etcd and control plane health checks
+KubeadmControlPlane rollout operations rely on [scale up](#scale up) and [scale down](#scale_down) which are be blocked based on Etcd and control plane health checks
   - See [Health checks](#Health checks) below.
 
+KubeadmControlPlane rollout is triggered by:
+  - Changes to Version
+  - Changes to the kubeadmConfigSpec
+  - Changes to the infrastructureRef
+
+##### KubeadmControlPlane rollout (using create-swap-and-delete)
+
 - The rollout algorithm is the following:
   - Find Machines that have an outdated spec
   - If there is a machine requiring rollout
     - Scale up control plane creating a machine with the new spec
     - Scale down control plane by removing one of the machine that needs rollout (the oldest out-of date machine in the failure domain that has the most control-plane machines on it)
-    
+
 - In order to determine if a Machine to be rolled out, KCP implements the following:
     - The infrastructureRef link used by each machine at creation time is stored in annotations at machine level.
     - The kubeadmConfigSpec used by each machine at creation time is stored in annotations at machine level.
         - If the annotation is not present (machine is either old or adopted), we won't roll out on any possible changes made in KCP's ClusterConfiguration given that we don't have enough information to make a decision.
            Users should use KCP.Spec.UpgradeAfter field to force a rollout in this case.
-    
+
+- The `upgradeAfter` field, which can be set to a specific time in the future
+  - Set to `nil` or the zero value of `time.Time` if no upgrades are desired
+  - An upgrade will run when that timestamp is passed
+  - Good for scheduling upgrades/SLOs
+  - Set `upgradeAfter` to now (in RFC3339 form) if an upgrade is required immediately
```

Comment:
This IMO should go up, as part of the least of things that can trigger a rollout

Reply:
Also, as a second thought, I think that the notes about how we determine when a machine needs rollout (L397-L401) and the not about tolerating manual node deletion (L409) should move to the previous paragraph because they are describing things shared between the two strategies